examples/bridge-app/linux/main.cpp

```diff
@@ -641,6 +655,82 @@ EmberAfStatus emberAfExternalAttributeWriteCallback(EndpointId endpoint, Cluster
     return ret;
 }
 
+void runOnOffRoomAction(Room * room, bool actionOn, EndpointId endpointId, uint16_t actionID, uint32_t invokeID, bool hasInvokeID)
+{
+    if (hasInvokeID)
+    {
+        BridgedActions::Events::StateChanged::Type event{ actionID, invokeID, BridgedActions::ActionStateEnum::kActive };
+        EventNumber eventNumber;
+        chip::app::LogEvent(event, endpointId, eventNumber);
+    }
+
+    // Check and run the action for ActionLight1 - ActionLight4
+    if (room->getName().compare(ActionLight1.GetLocation()) == 0)
+    {
+        ActionLight1.SetOnOff(actionOn);
+    }
+    if (room->getName().compare(ActionLight2.GetLocation()) == 0)
+    {
+        ActionLight2.SetOnOff(actionOn);
+    }
+    if (room->getName().compare(ActionLight3.GetLocation()) == 0)
+    {
+        ActionLight3.SetOnOff(actionOn);
+    }
+    if (room->getName().compare(ActionLight4.GetLocation()) == 0)
+    {
+        ActionLight4.SetOnOff(actionOn);
+    }
+
+    if (hasInvokeID)
+    {
+        BridgedActions::Events::StateChanged::Type event{ actionID, invokeID, BridgedActions::ActionStateEnum::kInactive };
+        EventNumber eventNumber;
+        chip::app::LogEvent(event, endpointId, eventNumber);
+    }
+}
+
+bool emberAfBridgedActionsClusterInstantActionCallback(app::CommandHandler * commandObj,
+                                                       const app::ConcreteCommandPath & commandPath,
+                                                       const BridgedActions::Commands::InstantAction::DecodableType & commandData)
+{
+    bool hasInvokeID      = false;
+    uint32_t invokeID     = 0;
+    EndpointId endpointID = commandPath.mEndpointId;
+    auto & actionID       = commandData.actionID;
+
+    if (commandData.invokeID.HasValue())
+    {
+        hasInvokeID = true;
+        invokeID    = commandData.invokeID.Value();
+    }
+
+    if (actionID == action1.getActionId() && action1.getIsVisible() && room1.getIsVisible())
```

Comment:
But the action is present in the action list even if the room is not visible?  That seems a little odd.

Reply:
Good point. I changed it so that if it is in the list it will run the action.